discovery/premature_cache.go

```diff
@@ -0,0 +1,266 @@
+package discovery
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/lightninglabs/neutrino/cache"
+	"github.com/lightninglabs/neutrino/cache/lru"
+	"github.com/lightningnetwork/lnd/lnwire"
+)
+
+// maxPrematureUpdates tracks the max amount of premature channel updates and
+// node announcements that we'll hold onto for each sender.
+const maxPrematureUpdates = 10
```

Comment:
Still getting through, but IMO this also needs to still retain the global limit, as otherwise we'll cache all the `NodeAnnouncements` which can be created essentially at no cost. Also see the other comment re the neutrino PR that makes a proper `Cache[K comparable, V cache.Value]` type.